### ISSUE TITLE
Add constant learning rate controller with patience

### ISSUE THREAD

**Issue by davidlmorton** [davidlmorton/pytorch-sconce#13]
Patience is how long the rate controller will continue returning new learning rate values based on the trajectory of the losses.  If the losses are not continuing to go down after a while you want to end training.  If you continue training you might overfit the model leading to worse generalization than if you had stopped training earlier.
